wsgi.py

```diff
@@ -812,13 +812,25 @@ def write_html_head(doc: yattag.doc.Doc, title: str) -> None:
     """Produces the <head> tag and its contents."""
     prefix = config.Config.get_uri_prefix()
     with doc.tag("head"):
+        doc.stag("meta", charset="UTF-8")
+        doc.stag("meta", name="viewport", content="width=device-width, initial-scale=1")
         with doc.tag("title"):
             doc.text(_("Where to map?") + title)
-        doc.stag("meta", charset="UTF-8")
-        doc.stag("link", rel="stylesheet", type="text/css", href=prefix + "/static/osm.css")
+        doc.stag("link", rel="icon", type="image/vnd.microsoft.icon", sizes="16x12", href=prefix + "/favicon.ico")
+        doc.stag("link", rel="icon", type="image/svg+xml", sizes="any", href=prefix + "/favicon.svg")
+
+        css_path = os.path.join(config.Config.get_workdir(), "osm.min.css")
+        with open(css_path, "r") as stream:
+            with doc.tag("style"):
+                doc.text(stream.read())
+
+        with doc.tag("noscript"):
```

Comment:
A HACKING.adoc ("codeing style") azt javasolja, hogy ne legyenek noscript tagek, helyette a python olyan htmlt generál ami js mentes, és utána a js már tudja módosítani a domot úgy, ahogy az neki kell. Ezt figyelembe tudnád venni itt is?

Jsből mikor betöltöd a chart kódot, akkor egy css selectorral le tudnád kérni a canvasokat és a display-none-jukat átírni.

Reply:
Olvastam a fájlban ezt a részt, de attól nem biztos, hogy egyet is értek vele. A `js`/`no-js` CSS konvenciót más library-ben is használják mások. Előnye, hogy nem villog a weboldal a JavaScript lefutásakor mivel ez a headben van, így már az első megjelenítés akkurátus lesz. A még nem aktív elemek is megjelennek, csak nem lesznek interaktívak a JavaScript letöltéséig, szerintem a progressive enhancement pont ezt mondja.

Ekvivalens alternatívaként `<noscript>` helyett lehetne ugyanez a blokk egy soros JavaScript ami kiüti/átírja a CSS-t mint az állatok, de ez egyrészt lassabb, másrészt nem világos, hogy mivel lenne jobb egyáltalán jobb (és amúgy sem illik `<head>`-ben JavaScriptezni ha nem muszáj).

Amúgy nem most akartam megcsinálni minden korábbi problémát egyszerre. Főleg, hogy arról beszéltünk, hogy a közeljövőben kikerül a canvasos kód.

Reply:
> mint az állatok

Kérlek ezt a stílust kerüld itt. Ez itt nem a matrix, ahol bármit megengedhetsz magadnak és akit ez zavar, majd ignore-ra rak.

Reply:
> Olvastam a fájlban ezt a részt, de attól nem biztos, hogy egyet is értek vele.

Esetleg lehetne olyat, hogy módosítod a coding style-t, hogy noscriptet csak feltételes CSS céljára kéretik használni.

Reply:
Bocsánat, ha félreértetted, a "mint az állatok" a mi szakmai köreinkben azt jelenti, hogy "erőből", "kézzel', "nyersen".

Reply:
Kész, részleteztem a `HACKING.adoc`-ban.